### PR TITLE
refactor: posts services

### PR DESCRIPTION
**Pull request by ivanmosa27**
This refactor change: 
-The routes naming.
-Implement the posts services following the SOLID principes.

### CODE REVIEW

File: Controllers/PostsController.cs [hunk 1]
```diff
@@ -9,113 +10,63 @@
 
 namespace blog_net_core.Controllers
 {
-    [Route("api/[controller]")]
+    [Route("[controller]")]
```
Comment (danielhdezller) [inline Controllers/PostsController.cs:13]: here should be "posts"

File: Controllers/PostsController.cs [hunk 2]
```diff
@@ -9,113 +10,63 @@
...
     public class PostsController : ControllerBase
     {
-        private BLOGContext _context;
-        public PostsController(BLOGContext context)
+        private readonly IBlogService _blogService;
+
+        public PostsController(IBlogService blogService)
         {
-            _context = context;
+            _blogService = blogService;
```
Comment (danielhdezller) [inline Controllers/PostsController.cs]: its the postService, the blog could be an entity that manage the relations of many posts

File: Properties/launchSettings.json [hunk 3]
```diff
@@ -12,15 +12,15 @@
     "IIS Express": {
       "commandName": "IISExpress",
       "launchBrowser": true,
-      "launchUrl": "api/posts/getall",
+      "launchUrl": "posts",
```
Comment (danielhdezller) [inline Properties/launchSettings.json]: the launchUrl should not be the controller route

File: Services/blogService.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace blog_net_core.Services{
+
+    public class blogService : IBlogService{
+
+        private readonly BLOGContext _context;
```
Comment (danielhdezller) [inline Services/blogService.cs]: It's not BLOG always use camel case, it's the PostsContext

File: Startup.cs [hunk 5]
```diff
@@ -28,6 +29,7 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers();
+            services.AddTransient<IBlogService, blogService>();
```
Comment (danielhdezller) [inline Startup.cs]: IPostsService

**Review by danielhdezller — CHANGES_REQUESTED** [review]
great job!

**Comment by ivanmosa27** [comment]
Closed

**Pull request closed without merging**